simularium/simularium-viewer
# Fix/color change bug

Problem
=======
in working on the color picker @interim17 and @ShrimpCryptid noticed a bug with wrong color being applied

Solution
========
The fix was to check if the color changes coming into the app had been reset to defaults. But in debugging we found that the color handling was confusing. There are two different color arrays: an array where each item is a single color, and `colorsData` where each color is represented by 4 numbers, so it's 4 times as long as the actual number of colors. We identify colors by their index in the first array, but sometimes need to switch between the two different indexes, and in each case they were labeled something like `colorIndex` or just `index`. Here I renamed everything as either a `colorId` :index into the shorter array where each item is a color and `colorDataIndex`: index into the 4 timex longer array. Hopefully this will reduce future confusion. 

So the majority of this changeset is documentation and tests. 

with @interim17 

## Type of change
Please delete options that are not relevant.

* Bug fix (non-breaking change which fixes an issue)
* This change requires a documentation update
* This change requires updated or new tests

Steps to Verify:
----------------
1. `npm start`
2. select a model 
3. change colors, add new colors, etc, should be stable.
